electron/main.ts

```diff
@@ -348,23 +362,29 @@ ipcMain.handle('install', async (event, args) => {
     command = `${legendaryBin} install ${game} --base-path ${defaultInstallPath} ${workers} -y |& tee ${logPath}`
   }
   console.log(`Installing ${game} with:`, command)
-  const noSpaceMsg = 'Not enough available disk space'
-  return await execAsync(command, { shell: '/bin/bash' })
-    .then(() => console.log('finished installing'))
-    .catch(async () => {
-      try {
-        const { stdout } = await execAsync(
-          `tail ${logPath} | grep 'disk space'`
-        )
+  try {
+    await execAsync(command, { shell: '/bin/bash' })
+    console.log('finished installing')
+  } catch (error) {
+    const noSpaceMsg = 'Not enough available disk space'
+    const genericMessage = 'installation canceled or had some error'
+    await execAsync(`tail ${logPath} | grep 'disk space'`)
+      .then(({ stdout }) => {
         if (stdout.includes(noSpaceMsg)) {
           console.log(noSpaceMsg)
-          return stdout
+          return showErrorBox(
+            i18next.t('box.error.diskspace.title', 'No Space'),
+            i18next.t(
+              'box.error.diskspace.message',
+              'Not enough available disk space'
+            )
+          )
         }
-        console.log('installation canceled or had some error')
-      } catch (err) {
-        return err
-      }
-    })
+        console.log(genericMessage)
+        return genericMessage
+      })
+      .catch(() => console.log(genericMessage))
```

Comment:
Maybe show the error within an error box, so if something goes wrong, the user at least sees that something went wrong.

Reply:
Also it looks like `genericMessage` is not translated.

Reply:
So, I was about to do that when I realized that this message is being shown everytime the install is canceled, and will be annoying I guess. Because Heroic sends a kill signal to Legendary to stop the installation, and then the exec command will throw an error and then the generic message will be shown. So I'm not sure if this will be good UX wise. 🤔